Upgrade Python to 3.8

seallard opened the pull request:
## Description
All the package versions we use support Python 3.8, so we can upgrade it from 3.7. The package can be installed and all tests pass with Python 3.8 locally at least.

It is necessary to upgrade the `S_main` and `P_main` conda environments to use Python 3.8 as well. This can be done in stage like this:
```bash
conda activate S_main
conda install python=3.8
python --version
```
Need to go through housekeeper and any other tools running in the `S_main`/`P_main` environments before merging this.

### Fixed
- Upgrade to Python 3.8 from 3.7

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

seallard commented:
Package versions
[] alembic             1.11.1
[] annotated-types     0.5.0
[] anyio               3.7.1
[] atomicwrites        1.4.1
[] attrs               23.1.0
[] bcrypt              4.0.1
[] blinker             1.6.2
[] CacheControl        0.13.1
[] cachetools          5.3.1
[] certifi             2023.5.7
[] cffi                1.16.0
[] cfgv                3.4.0
[] cgmodels            0.11.0
[] charset-normalizer  3.3.0
[] click               6.7
[] colorama            0.4.6
[] coloredlogs         15.0.1
[] coverage            6.5.0
[] coveralls           3.3.1
[] cryptography        40.0.2
[] dataclasses         0.6
[] distlib             0.3.7
[] dnspython           2.4.2
[] docopt              0.6.2
[] email-validator     2.0.0.post2
[] et-xmlfile          1.1.0
[] exceptiongroup      1.1.3
[] fastapi             0.103.2
[] filelock            3.12.4
[] Flask               1.1.4
[] Flask-Admin         1.6.1
[] Flask-Cors          4.0.0
[] Flask-Dance         5.1.0
[] Flask-WTF           1.2.1
[] genologics          0.4.6
[] google-auth         2.18.1
[] gunicorn            21.2.0
[] h11                 0.14.0
[] housekeeper         4.8.5
[] httptools           0.6.0
[] humanfriendly       10.0
[] identify            2.5.30
[] idna                3.4
[] importlib-metadata  6.6.0
[] importlib-resources 6.1.0
[] iniconfig           2.0.0
[] itsdangerous        1.1.0
[] Jinja2              2.11.3
[] lxml                4.9.3
[] Mako                1.2.4
[] markdown-it-py      2.2.0
[] MarkupSafe          2.0.1
[] marshmallow         3.20.1
[] mdurl               0.1.2
[] mock                5.1.0
[] more-itertools      10.1.0
[] msgpack             1.0.7
[] nodeenv             1.8.0
[] numpy               1.24.3
[] oauthlib            3.2.2
[] openpyxl            3.0.10
[] packaging           23.1
[] pandas              2.0.1
[] paramiko            3.3.1
[] petname             2.6
[] pip                 23.2.1
[] platformdirs        3.11.0
[] pluggy              1.3.0
[] pre-commit          3.4.0
[] py                  1.11.0
[] pyasn1              0.5.0
[] pyasn1-modules      0.3.0
[] pycparser           2.21
[] pydantic            2.3.0
[] pydantic_core       2.6.3
[] pydash              7.0.3
[] Pygments            2.15.1
[] PyMySQL             1.0.3
[] PyNaCl              1.5.0
[] pyparsing           3.1.1
[] pytest              7.3.1
[] pytest-cov          4.1.0
[] pytest-mock         3.11.1
[] python-dateutil     2.8.2
[] python-multipart    0.0.6
[] pytz                2023.3
[] PyYAML              6.0.1
[] requests            2.30.0
[] requests-oauthlib   1.3.1
[] rich                13.3.5
[] rsa                 4.9
[] ruff                0.0.292
[] sendmail-container  0.2.0
[] setuptools          56.0.0
[] six                 1.16.0
[] sniffio             1.3.0
[] SQLAlchemy          1.3.24
[] starlette           0.27.0
[] tabulate            0.9.0
[] tomli               2.0.1
[] typing_extensions   4.8.0
[] tzdata              2023.3
[] urllib3             1.26.17
[] URLObject           2.4.3
[] uvicorn             0.5.1
[] uvloop              0.17.0
[] virtualenv          20.24.5
[] wcwidth             0.2.8
[] websockets          11.0.3
[] Werkzeug            1.0.1
[] WTForms             3.0.1
zipp                3.17.0

seallard commented:
Package versions
- [ ] alembic             1.11.1
- [ ] annotated-types     0.5.0
- [ ] anyio               3.7.1
- [ ] atomicwrites        1.4.1
- [ ] attrs               23.1.0
- [ ] bcrypt              4.0.1
- [ ] blinker             1.6.2
- [ ] CacheControl        0.13.1
- [ ] cachetools          5.3.1
- [ ] certifi             2023.5.7
- [ ] cffi                1.16.0
- [ ] cfgv                3.4.0
- [ ] cgmodels            0.11.0
- [ ] charset-normalizer  3.3.0
- [ ] click               6.7
- [ ] colorama            0.4.6
- [ ] coloredlogs         15.0.1
- [ ] coverage            6.5.0
- [ ] coveralls           3.3.1
- [ ] cryptography        40.0.2
- [ ] dataclasses         0.6
- [ ] distlib             0.3.7
- [ ] dnspython           2.4.2
- [ ] docopt              0.6.2
- [ ] email-validator     2.0.0.post2
- [ ] et-xmlfile          1.1.0
- [ ] exceptiongroup      1.1.3
- [ ] fastapi             0.103.2
- [ ] filelock            3.12.4
- [ ] Flask               1.1.4
- [ ] Flask-Admin         1.6.1
- [ ] Flask-Cors          4.0.0
- [ ] Flask-Dance         5.1.0
- [ ] Flask-WTF           1.2.1
- [ ] genologics          0.4.6
- [ ] google-auth         2.18.1
- [ ] gunicorn            21.2.0
- [ ] h11                 0.14.0
- [ ] housekeeper         4.8.5
- [ ] httptools           0.6.0
- [ ] humanfriendly       10.0
- [ ] identify            2.5.30
- [ ] idna                3.4
- [ ] importlib-metadata  6.6.0
- [ ] importlib-resources 6.1.0
- [ ] iniconfig           2.0.0
- [ ] itsdangerous        1.1.0
- [ ] Jinja2              2.11.3
- [ ] lxml                4.9.3
- [ ] Mako                1.2.4
- [ ] markdown-it-py      2.2.0
- [ ] MarkupSafe          2.0.1
- [ ] marshmallow         3.20.1
- [ ] mdurl               0.1.2
- [ ] mock                5.1.0
- [ ] more-itertools      10.1.0
- [ ] msgpack             1.0.7
- [ ] nodeenv             1.8.0
- [ ] numpy               1.24.3
- [ ] oauthlib            3.2.2
- [ ] openpyxl            3.0.10
- [ ] packaging           23.1
- [ ] pandas              2.0.1
- [ ] paramiko            3.3.1
- [ ] petname             2.6
- [ ] pip                 23.2.1
- [ ] platformdirs        3.11.0
- [ ] pluggy              1.3.0
- [ ] pre-commit          3.4.0
- [ ] py                  1.11.0
- [ ] pyasn1              0.5.0
- [ ] pyasn1-modules      0.3.0
- [ ] pycparser           2.21
- [ ] pydantic            2.3.0
- [ ] pydantic_core       2.6.3
- [ ] pydash              7.0.3
- [ ] Pygments            2.15.1
- [ ] PyMySQL             1.0.3
- [ ] PyNaCl              1.5.0
- [ ] pyparsing           3.1.1
- [ ] pytest              7.3.1
- [ ] pytest-cov          4.1.0
- [ ] pytest-mock         3.11.1
- [ ] python-dateutil     2.8.2
- [ ] python-multipart    0.0.6
- [ ] pytz                2023.3
- [ ] PyYAML              6.0.1
- [ ] requests            2.30.0
- [ ] requests-oauthlib   1.3.1
- [ ] rich                13.3.5
- [ ] rsa                 4.9
- [ ] ruff                0.0.292
- [ ] sendmail-container  0.2.0
- [ ] setuptools          56.0.0
- [ ] six                 1.16.0
- [ ] sniffio             1.3.0
- [ ] SQLAlchemy          1.3.24
- [ ] starlette           0.27.0
- [ ] tabulate            0.9.0
- [ ] tomli               2.0.1
- [ ] typing_extensions   4.8.0
- [ ] tzdata              2023.3
- [ ] urllib3             1.26.17
- [ ] URLObject           2.4.3
- [ ] uvicorn             0.5.1
- [ ] uvloop              0.17.0
- [ ] virtualenv          20.24.5
- [ ] wcwidth             0.2.8
- [ ] websockets          11.0.3
- [ ] Werkzeug            1.0.1
- [ ] WTForms             3.0.1
- [ ] zipp                3.17.0

ChrOertlin commented:
Locally I run 3.9 and tests work as well. Maybe we can try the latest possible?

Vince-janv commented:
Nice work @seallard! Good point regarding the S_main environment. I took a quick look at https://github.com/Clinical-Genomics/servers/blob/ffa922eaac6cea31f8b3147cb0a5948169a07a0c/resources/hasta.scilifelab.se/usestage.sh. And it seems like the affected apps are indeed `cg`, `housekeeper` and `paxa`.

@ChrOertlin regarding going directly to 3.9 I think it's safest to take it one minor at a time. We know that our unit tests don't catch everything 😬

seallard commented:
Package versions
- [ ] housekeeper         4.8.5  -
- [ ] cgmodels            0.11.0  -
- [ ] genologics          0.4.6  - 
- [ ] setuptools          56.0.0  - [requires > 3.6](https://pypi.org/project/setuptools/56.0.0/)
- [ ] petname             2.6  - does not say
- [x] alembic             1.11.1 - [supports 3.7 and above](https://alembic.sqlalchemy.org/en/latest/front.html#dependencies)
- [x] blinker             1.6.2 - [supports 3.7 and above](https://pypi.org/project/blinker/)
- [x] CacheControl        0.13.1 - [supports 3.7 and above](https://pypi.org/project/CacheControl/)
- [x] cachetools          5.3.1 - [supports 3.7 and above](https://pypi.org/project/cachetools/)
- [x] click               6.7  - cannot find support, should work
- [x] Flask               1.1.4  - [supports 3.7 and above](https://pypi.org/project/Flask/1.1.4/)
- [ ] Flask-Admin         1.6.1  - [supports 3.7 and above](https://pypi.org/project/Flask-Admin/1.6.1/)
- [ ] Flask-Cors          4.0.0  - [supports 3.7 and above](https://pypi.org/project/Flask-Cors/)
- [ ] Flask-Dance         5.1.0  - [supports 3.7 and above](https://pypi.org/project/Flask-Dance/5.1.0/)
- [ ] Flask-WTF           1.2.1  - not stated
- [ ] google-auth         2.18.1  - [supports 3.7 and above](https://pypi.org/project/google-auth/2.18.1/)
- [ ] gunicorn            21.2.0  - [supports 3.7 and above](https://pypi.org/project/gunicorn/21.2.0/)
- [ ] Jinja2              2.11.3  - [supports 3.8](https://pypi.org/project/Jinja2/2.11.3/)
- [ ] lxml                4.9.3  - [supports 3.7 and above](https://pypi.org/project/lxml/4.9.3/)
- [ ] MarkupSafe          2.0.1  - [supports 3.6 and above](https://pypi.org/project/MarkupSafe/2.0.1/)
- [ ] marshmallow         3.20.1  - [supports 3.7 and above](https://pypi.org/project/marshmallow/)
- [ ] numpy               1.24.3  - [supports 3.7 and above](https://pypi.org/project/numpy/1.24.3/)
- [ ] openpyxl            3.0.10  - [supports 3.7 and above](https://pypi.org/project/openpyxl/3.0.10/)
- [ ] pandas              2.0.1  - [supports 3.7 and above](https://pypi.org/project/pandas/)
- [ ] paramiko            3.3.1  - [supports 3.7 and above](https://pypi.org/project/paramiko/3.3.1/)
- [ ] pydantic            2.3.0 - [supports 3.7 and above](https://pypi.org/project/pydantic/2.3.0/)
- [ ] python-dateutil     2.8.2  - [supports 3.7 and above](https://pypi.org/project/python-dateutil/)
- [ ] PyYAML              6.0.1  - [supports 3.7 and above](https://pypi.org/project/PyYAML/)
- [ ] requests            2.30.0  - [supports 3.7 and above](https://pypi.org/project/requests/)
- [ ] SQLAlchemy          1.3.24  - [supports 3.7 and above](https://pypi.org/project/SQLAlchemy/1.3.24/)
- [ ] tabulate            0.9.0  - [supports 3.7 and above](https://pypi.org/project/tabulate/)
- [ ] tomli               2.0.1  - [supports 3.7 and above](https://pypi.org/project/tomli/)
- [ ] urllib3             1.26.17  - [supports 3.7 and above](https://pypi.org/project/urllib3/1.26.17/)
- [ ] Werkzeug            1.0.1  - [supports 3.8](https://pypi.org/project/Werkzeug/1.0.1/)
- [ ] WTForms             3.0.1  - [supports 3.7 and above](https://pypi.org/project/WTForms/3.0.1/)

diitaz93 commented:
We could deprecate cgmodels first so we don't have to worry about it @seallard. Although this can be done before, from cgmodels we only import str enum classes

seallard commented:
> We could deprecate cgmodels first so we don't have to worry about it @seallard. Although this can be done before, from cgmodels we only import str enum classes

No worries! Should be fine 👍

seallard commented:
Installing Python 3.8 in stage:
```
θ66° [sebastian.allard@hasta:~] [S_base] $ conda activate S_main
[sebastian.allard@hasta:~] [S_main] 5s $ conda install python=3.8
Collecting package metadata (current_repodata.json): done
Solving environment: \ 
The environment is inconsistent, please check the package plan carefully
The following packages are causing the inconsistency:

  - conda-forge/noarch::wheel==0.37.1=pyhd8ed1ab_0
  - defaults/linux-64::python==3.7.1=h0371630_3
  - conda-forge/noarch::pip==22.2.2=pyhd8ed1ab_0
  - conda-forge/linux-64::python_abi==3.7=2_cp37m
done


==> WARNING: A newer version of conda exists. <==
  current version: 4.12.0
  latest version: 23.9.0

Please update conda by running

    $ conda update -n base -c defaults conda



## Package Plan ##

  environment location: /home/proj/stage/bin/miniconda3/envs/S_main

  added / updated specs:
    - python=3.8


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    python-3.8.17              |he550d4f_0_cpython        23.4 MB  conda-forge
    python_abi-3.8             |           4_cp38           6 KB  conda-forge
    ------------------------------------------------------------
                                           Total:        23.4 MB

The following NEW packages will be INSTALLED:

  bzip2              conda-forge/linux-64::bzip2-1.0.8-h7f98852_4
  ld_impl_linux-64   conda-forge/linux-64::ld_impl_linux-64-2.40-h41732ed_0
  libnsl             conda-forge/linux-64::libnsl-2.0.0-hd590300_1
  libsqlite          conda-forge/linux-64::libsqlite-3.43.0-h2797004_0
  libuuid            conda-forge/linux-64::libuuid-2.38.1-h0b41bf4_0
  setuptools         conda-forge/noarch::setuptools-68.2.2-pyhd8ed1ab_0

The following packages will be REMOVED:

  sqlite-3.28.0-h8b20d00_0
  zlib-1.2.12-h166bdaf_2

The following packages will be UPDATED:

  ca-certificates                      2022.6.15-ha878542_0 --> 2023.7.22-hbcca054_0
  libffi                                3.2.1-he1b5a44_1007 --> 3.4.2-h7f98852_5
  libzlib                                 1.2.12-h166bdaf_2 --> 1.2.13-hd590300_5
  ncurses                                    6.3-h27087fc_1 --> 6.4-hcb278e6_0
  openssl                                 1.1.1q-h166bdaf_0 --> 3.1.3-hd590300_0
  python                 pkgs/main::python-3.7.1-h0371630_3 --> conda-forge::python-3.8.17-he550d4f_0_cpython
  python_abi                                    3.7-2_cp37m --> 3.8-4_cp38
  readline                                7.0-hf8c457e_1001 --> 8.2-h8228510_1


Proceed ([y]/n)? y


Downloading and Extracting Packages
python-3.8.17        | 23.4 MB   | ################################################################################################################################################################################################################################################ | 100% 
python_abi-3.8       | 6 KB      | ################################################################################################################################################################################################################################################ | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
```

seallard commented:
Deploying this branch:
```
[sebastian.allard@hasta:~] [S_main] 3m3s $ bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b upgrade-python
ERROR conda.cli.main_run:execute(41): `conda run /home/proj/stage/bin/miniconda3/envs/S_main/bin/cg --version` failed. (See above for error)
/home/sebastian.allard/tmp/tmptqz5eo7p: /home/proj/stage/bin/miniconda3/envs/S_main/bin/cg: /home/proj/stage/bin/miniconda3/envs/S_main/bin/python3.7: bad interpreter: No such file or directory

Collecting git+https://github.com/Clinical-Genomics/cg@upgrade-python
  Running command git clone --quiet https://github.com/Clinical-Genomics/cg /home/sebastian.allard/tmp/pip-req-build-9m0lmrvd
  Cloning https://github.com/Clinical-Genomics/cg (to revision upgrade-python) to ./tmp/pip-req-build-9m0lmrvd
  Running command git checkout -b upgrade-python --track origin/upgrade-python
  Resolved https://github.com/Clinical-Genomics/cg to commit 2613925595faad73950c2d342620c543bfeffb3e
  Switched to a new branch 'upgrade-python'
  Installing build dependencies: started
  Branch upgrade-python set up to track remote branch upgrade-python from origin.
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Building wheels for collected packages: cg
  Building wheel for cg (pyproject.toml): started
  Building wheel for cg (pyproject.toml): finished with status 'done'
  Created wheel for cg: filename=cg-51.2.2-py3-none-any.whl size=740235 sha256=56d2595a3d56e2759e56940914d9575d3a3a73686f4bb9e61f002b108c84a748
  Stored in directory: /home/sebastian.allard/tmp/pip-ephem-wheel-cache-gnmkxgo6/wheels/68/b3/1a/e11f5209f2c79abbefd0f1ef6a7c0845519466dc75a391ef29
Successfully built cg
Installing collected packages: cg
Successfully installed cg-51.2.2
Collecting git+https://github.com/Clinical-Genomics/cg@upgrade-python
  Running command git clone --quiet https://github.com/Clinical-Genomics/cg /home/sebastian.allard/tmp/pip-req-build-npgfs2zn
  Cloning https://github.com/Clinical-Genomics/cg (to revision upgrade-python) to ./tmp/pip-req-build-npgfs2zn
  Running command git checkout -b upgrade-python --track origin/upgrade-python
  Resolved https://github.com/Clinical-Genomics/cg to commit 2613925595faad73950c2d342620c543bfeffb3e
  Switched to a new branch 'upgrade-python'
  Installing build dependencies: started
  Branch upgrade-python set up to track remote branch upgrade-python from origin.
  Installing build dependencies: finished with status 'done'
WARNING: pydantic 2.4.2 does not provide the extra 'dotenv'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting gunicorn
  Downloading gunicorn-21.2.0-py3-none-any.whl (80 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 80.2/80.2 kB 4.9 MB/s eta 0:00:00
Collecting cachetools
  Downloading cachetools-5.3.1-py3-none-any.whl (9.3 kB)
Collecting google-auth
  Downloading google_auth-2.23.2-py2.py3-none-any.whl (181 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 182.0/182.0 kB 20.5 MB/s eta 0:00:00
Collecting pydantic>2.0
  Downloading pydantic-2.4.2-py3-none-any.whl (395 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 395.8/395.8 kB 47.8 MB/s eta 0:00:00
Requirement already satisfied: packaging in /home/proj/stage/bin/miniconda3/envs/S_main/bin (from cg==51.2.2) (21.3)
Collecting genologics
  Downloading genologics-0.4.6.tar.gz (31 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting SQLAlchemy<1.4
  Downloading SQLAlchemy-1.3.24-cp38-cp38-manylinux2010_x86_64.whl (1.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.3/1.3 MB 46.8 MB/s eta 0:00:00
Collecting openpyxl==3.0.10
  Downloading openpyxl-3.0.10-py2.py3-none-any.whl (242 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 242.1/242.1 kB 37.7 MB/s eta 0:00:00
Collecting python-dateutil
  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 247.7/247.7 kB 24.3 MB/s eta 0:00:00
Collecting pymysql
  Downloading PyMySQL-1.1.0-py3-none-any.whl (44 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 44.8/44.8 kB 6.1 MB/s eta 0:00:00
Collecting Flask-CORS
  Downloading Flask_Cors-4.0.0-py2.py3-none-any.whl (14 kB)
Collecting urllib3
  Downloading urllib3-2.0.6-py3-none-any.whl (123 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 123.8/123.8 kB 15.6 MB/s eta 0:00:00
Collecting requests[security]
  Downloading requests-2.31.0-py3-none-any.whl (62 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.6/62.6 kB 9.8 MB/s eta 0:00:00
Collecting cachecontrol
  Downloading cachecontrol-0.13.1-py3-none-any.whl (22 kB)
Collecting lxml
  Downloading lxml-4.9.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (7.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.1/7.1 MB 57.7 MB/s eta 0:00:00
Collecting markupsafe<2.1
  Downloading MarkupSafe-2.0.1-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (30 kB)
Collecting petname
  Downloading petname-2.6.tar.gz (8.4 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting cgmodels>=0.11.0
  Downloading cgmodels-0.11.0-py3-none-any.whl (9.2 kB)
Requirement already satisfied: marshmallow>3 in /home/proj/stage/bin/miniconda3/envs/S_main/bin (from cg==51.2.2) (3.18.0)
Collecting flask-wtf
  Downloading flask_wtf-1.2.1-py3-none-any.whl (12 kB)
Collecting alembic
  Downloading alembic-1.12.0-py3-none-any.whl (226 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 226.0/226.0 kB 32.4 MB/s eta 0:00:00
Collecting sendmail-container
  Downloading sendmail_container-0.2.0-py3-none-any.whl (4.3 kB)
Collecting blinker
  Downloading blinker-1.6.2-py3-none-any.whl (13 kB)
Collecting Jinja2
  Downloading Jinja2-3.1.2-py3-none-any.whl (133 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 133.1/133.1 kB 24.5 MB/s eta 0:00:00
Requirement already satisfied: setuptools>=39.2.0 in /home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.8/site-packages (from cg==51.2.2) (68.2.2)
Collecting Flask-Dance
  Downloading flask_dance-7.0.0-py3-none-any.whl (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.0/62.0 kB 13.8 MB/s eta 0:00:00
Collecting werkzeug
  Downloading werkzeug-3.0.0-py3-none-any.whl (226 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 226.6/226.6 kB 28.9 MB/s eta 0:00:00
Collecting paramiko
  Downloading paramiko-3.3.1-py3-none-any.whl (224 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 224.8/224.8 kB 33.6 MB/s eta 0:00:00
Collecting typing-extensions
  Downloading typing_extensions-4.8.0-py3-none-any.whl (31 kB)
Collecting housekeeper<5.0,>=4.6
  Downloading housekeeper-4.8.5-py2.py3-none-any.whl (31 kB)
Collecting Click<7.0
  Downloading click-6.7-py2.py3-none-any.whl (71 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 71.2/71.2 kB 14.2 MB/s eta 0:00:00
Collecting pandas
  Downloading pandas-2.0.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.4 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.4/12.4 MB 56.5 MB/s eta 0:00:00
Collecting pyyaml
  Downloading PyYAML-6.0.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (736 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 736.6/736.6 kB 76.7 MB/s eta 0:00:00
Collecting Flask-Admin
  Downloading Flask_Admin-1.6.1-py3-none-any.whl (7.5 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.5/7.5 MB 71.3 MB/s eta 0:00:00
Collecting tabulate
  Downloading tabulate-0.9.0-py3-none-any.whl (35 kB)
Collecting coloredlogs
  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 46.0/46.0 kB 8.8 MB/s eta 0:00:00
Collecting Flask>=1.1.1
  Downloading flask-3.0.0-py3-none-any.whl (99 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 99.7/99.7 kB 20.6 MB/s eta 0:00:00
Collecting et-xmlfile
  Using cached et_xmlfile-1.1.0-py3-none-any.whl (4.7 kB)
Collecting coverage
  Downloading coverage-7.3.2-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (228 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 228.6/228.6 kB 27.6 MB/s eta 0:00:00
Collecting more-itertools
  Downloading more_itertools-10.1.0-py3-none-any.whl (55 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.8/55.8 kB 12.2 MB/s eta 0:00:00
Collecting colorama
  Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Requirement already satisfied: pyparsing in /home/proj/stage/bin/miniconda3/envs/S_main/bin (from cgmodels>=0.11.0->cg==51.2.2) (3.0.9)
Collecting pytest
  Downloading pytest-7.4.2-py3-none-any.whl (324 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 324.5/324.5 kB 34.2 MB/s eta 0:00:00
Collecting wcwidth
  Downloading wcwidth-0.2.8-py2.py3-none-any.whl (31 kB)
Requirement already satisfied: zipp in /home/proj/stage/bin/miniconda3/envs/S_main/bin (from cgmodels>=0.11.0->cg==51.2.2) (3.8.1)
Collecting pytest-cov
  Downloading pytest_cov-4.1.0-py3-none-any.whl (21 kB)
Collecting py
  Downloading py-1.11.0-py2.py3-none-any.whl (98 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.7/98.7 kB 18.0 MB/s eta 0:00:00
Collecting atomicwrites
  Downloading atomicwrites-1.4.1.tar.gz (14 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Requirement already satisfied: importlib-metadata in /home/proj/stage/bin/miniconda3/envs/S_main/bin (from cgmodels>=0.11.0->cg==51.2.2) (5.0.0)
Collecting attrs
  Downloading attrs-23.1.0-py3-none-any.whl (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.2/61.2 kB 9.8 MB/s eta 0:00:00
Collecting pluggy
  Downloading pluggy-1.3.0-py3-none-any.whl (18 kB)
Collecting dataclasses
  Downloading dataclasses-0.6-py3-none-any.whl (14 kB)
Collecting Flask>=1.1.1
  Downloading flask-2.3.3-py3-none-any.whl (96 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.1/96.1 kB 15.0 MB/s eta 0:00:00
  Downloading Flask-2.3.2-py3-none-any.whl (96 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.9/96.9 kB 18.9 MB/s eta 0:00:00
  Downloading Flask-2.3.1-py3-none-any.whl (96 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.0/97.0 kB 20.3 MB/s eta 0:00:00
Collecting itsdangerous>=2.1.2
  Downloading itsdangerous-2.1.2-py3-none-any.whl (15 kB)
Collecting Flask>=1.1.1
  Downloading Flask-2.3.0-py3-none-any.whl (96 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.0/97.0 kB 18.7 MB/s eta 0:00:00
  Downloading Flask-2.2.5-py3-none-any.whl (101 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 101.8/101.8 kB 27.6 MB/s eta 0:00:00
  Downloading Flask-2.2.4-py3-none-any.whl (101 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 101.7/101.7 kB 21.1 MB/s eta 0:00:00
  Downloading Flask-2.2.3-py3-none-any.whl (101 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 101.8/101.8 kB 23.0 MB/s eta 0:00:00
  Downloading Flask-2.2.2-py3-none-any.whl (101 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 101.5/101.5 kB 19.3 MB/s eta 0:00:00
  Downloading Flask-2.2.1-py3-none-any.whl (101 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 101.5/101.5 kB 22.2 MB/s eta 0:00:00
  Downloading Flask-2.2.0-py3-none-any.whl (101 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 101.1/101.1 kB 21.8 MB/s eta 0:00:00
  Downloading Flask-2.1.3-py3-none-any.whl (95 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 95.6/95.6 kB 17.8 MB/s eta 0:00:00
  Downloading Flask-2.1.2-py3-none-any.whl (95 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 95.2/95.2 kB 14.7 MB/s eta 0:00:00
  Downloading Flask-2.1.1-py3-none-any.whl (95 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 95.2/95.2 kB 22.4 MB/s eta 0:00:00
  Downloading Flask-2.1.0-py3-none-any.whl (95 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 95.2/95.2 kB 20.3 MB/s eta 0:00:00
  Downloading Flask-2.0.3-py3-none-any.whl (95 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 95.6/95.6 kB 13.4 MB/s eta 0:00:00
  Downloading Flask-2.0.2-py3-none-any.whl (95 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 95.2/95.2 kB 23.1 MB/s eta 0:00:00
  Downloading Flask-2.0.1-py3-none-any.whl (94 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 94.8/94.8 kB 18.9 MB/s eta 0:00:00
  Downloading Flask-2.0.0-py3-none-any.whl (93 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 93.2/93.2 kB 19.4 MB/s eta 0:00:00
  Downloading Flask-1.1.4-py2.py3-none-any.whl (94 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 94.6/94.6 kB 14.4 MB/s eta 0:00:00
Collecting itsdangerous<2.0,>=0.24
  Downloading itsdangerous-1.1.0-py2.py3-none-any.whl (16 kB)
Collecting Jinja2
  Downloading Jinja2-2.11.3-py2.py3-none-any.whl (125 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 125.7/125.7 kB 26.8 MB/s eta 0:00:00
Collecting werkzeug
  Downloading Werkzeug-1.0.1-py2.py3-none-any.whl (298 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 298.6/298.6 kB 47.1 MB/s eta 0:00:00
Requirement already satisfied: rich in /home/proj/stage/bin/miniconda3/envs/S_main/bin (from housekeeper<5.0,>=4.6->cg==51.2.2) (12.6.0)
Collecting pydantic-core==2.10.1
  Downloading pydantic_core-2.10.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.1/2.1 MB 32.9 MB/s eta 0:00:00
Collecting annotated-types>=0.4.0
  Downloading annotated_types-0.5.0-py3-none-any.whl (11 kB)
Collecting Mako
  Downloading Mako-1.2.4-py3-none-any.whl (78 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.7/78.7 kB 13.2 MB/s eta 0:00:00
Collecting importlib-resources
  Downloading importlib_resources-6.1.0-py3-none-any.whl (33 kB)
Collecting msgpack>=0.5.2
  Downloading msgpack-1.0.7-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (534 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 534.8/534.8 kB 65.9 MB/s eta 0:00:00
Requirement already satisfied: humanfriendly>=9.1 in /home/proj/stage/bin/miniconda3/envs/S_main/bin (from coloredlogs->cg==51.2.2) (10.0)
Collecting wtforms
  Downloading WTForms-3.0.1-py3-none-any.whl (136 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 136.5/136.5 kB 22.2 MB/s eta 0:00:00
Collecting urlobject
  Downloading URLObject-2.4.3.tar.gz (27 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting requests-oauthlib>=1.0.0
  Downloading requests_oauthlib-1.3.1-py2.py3-none-any.whl (23 kB)
Collecting oauthlib>=3.2
  Downloading oauthlib-3.2.2-py3-none-any.whl (151 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 151.7/151.7 kB 24.7 MB/s eta 0:00:00
INFO: pip is looking at multiple versions of flask-dance to determine which version is compatible with other requirements. This could take a while.
Collecting Flask-Dance
  Downloading flask_dance-6.2.0-py3-none-any.whl (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.5/61.5 kB 12.3 MB/s eta 0:00:00
  Downloading Flask_Dance-6.1.1-py3-none-any.whl (59 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 59.2/59.2 kB 12.4 MB/s eta 0:00:00
  Downloading Flask_Dance-6.1.0-py2.py3-none-any.whl (59 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 59.2/59.2 kB 10.5 MB/s eta 0:00:00
  Downloading Flask_Dance-6.0.0-py2.py3-none-any.whl (58 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 59.0/59.0 kB 9.8 MB/s eta 0:00:00
  Downloading Flask_Dance-5.1.0-py2.py3-none-any.whl (59 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 59.5/59.5 kB 11.5 MB/s eta 0:00:00
Collecting pyasn1-modules>=0.2.1
  Downloading pyasn1_modules-0.3.0-py2.py3-none-any.whl (181 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 181.3/181.3 kB 31.3 MB/s eta 0:00:00
Collecting rsa<5,>=3.1.4
  Downloading rsa-4.9-py3-none-any.whl (34 kB)
Collecting tzdata>=2022.1
  Downloading tzdata-2023.3-py2.py3-none-any.whl (341 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 341.8/341.8 kB 48.6 MB/s eta 0:00:00
Collecting pytz>=2020.1
  Downloading pytz-2023.3.post1-py2.py3-none-any.whl (502 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 502.5/502.5 kB 67.1 MB/s eta 0:00:00
Collecting numpy>=1.20.3
  Downloading numpy-1.24.4-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 17.3/17.3 MB 38.3 MB/s eta 0:00:00
Collecting six>=1.5
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting pynacl>=1.5
  Downloading PyNaCl-1.5.0-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 94.0 MB/s eta 0:00:00
Collecting cryptography>=3.3
  Downloading cryptography-41.0.4-cp37-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.4 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.4/4.4 MB 98.2 MB/s eta 0:00:00
Collecting bcrypt>=3.2
  Downloading bcrypt-4.0.1-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (593 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 593.5/593.5 kB 65.1 MB/s eta 0:00:00
Collecting charset-normalizer<4,>=2
  Downloading charset_normalizer-3.3.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (137 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 137.9/137.9 kB 24.5 MB/s eta 0:00:00
Collecting certifi>=2017.4.17
  Downloading certifi-2023.7.22-py3-none-any.whl (158 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 158.3/158.3 kB 32.4 MB/s eta 0:00:00
Collecting idna<4,>=2.5
  Downloading idna-3.4-py3-none-any.whl (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.5/61.5 kB 12.2 MB/s eta 0:00:00
Collecting fastapi
  Downloading fastapi-0.103.2-py3-none-any.whl (66 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 66.3/66.3 kB 14.2 MB/s eta 0:00:00
Collecting email-validator
  Downloading email_validator-2.0.0.post2-py3-none-any.whl (31 kB)
Collecting uvicorn
  Downloading uvicorn-0.23.2-py3-none-any.whl (59 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 59.5/59.5 kB 13.2 MB/s eta 0:00:00
Collecting python-multipart
  Downloading python_multipart-0.0.6-py3-none-any.whl (45 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 45.7/45.7 kB 8.4 MB/s eta 0:00:00
Collecting cffi>=1.12
  Downloading cffi-1.16.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (444 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 444.7/444.7 kB 59.0 MB/s eta 0:00:00
Collecting pyasn1<0.6.0,>=0.4.6
  Downloading pyasn1-0.5.0-py2.py3-none-any.whl (83 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 83.9/83.9 kB 17.3 MB/s eta 0:00:00
Collecting dnspython>=2.0.0
  Downloading dnspython-2.4.2-py3-none-any.whl (300 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 300.4/300.4 kB 40.3 MB/s eta 0:00:00
Collecting starlette<0.28.0,>=0.27.0
  Downloading starlette-0.27.0-py3-none-any.whl (66 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 67.0/67.0 kB 12.5 MB/s eta 0:00:00
Collecting anyio<4.0.0,>=3.7.1
  Downloading anyio-3.7.1-py3-none-any.whl (80 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 80.9/80.9 kB 14.8 MB/s eta 0:00:00
Collecting iniconfig
  Downloading iniconfig-2.0.0-py3-none-any.whl (5.9 kB)
Collecting tomli>=1.0.0
  Downloading tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting exceptiongroup>=1.0.0rc8
  Downloading exceptiongroup-1.1.3-py3-none-any.whl (14 kB)
Requirement already satisfied: commonmark<0.10.0,>=0.9.0 in /home/proj/stage/bin/miniconda3/envs/S_main/bin (from rich->housekeeper<5.0,>=4.6->cg==51.2.2) (0.9.1)
Requirement already satisfied: pygments<3.0.0,>=2.6.0 in /home/proj/stage/bin/miniconda3/envs/S_main/bin (from rich->housekeeper<5.0,>=4.6->cg==51.2.2) (2.13.0)
Collecting uvicorn
  Downloading uvicorn-0.23.1-py3-none-any.whl (59 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 59.5/59.5 kB 10.7 MB/s eta 0:00:00
  Downloading uvicorn-0.23.0-py3-none-any.whl (59 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 59.6/59.6 kB 14.0 MB/s eta 0:00:00
  Downloading uvicorn-0.22.0-py3-none-any.whl (58 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.3/58.3 kB 12.1 MB/s eta 0:00:00
  Downloading uvicorn-0.21.1-py3-none-any.whl (57 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.8/57.8 kB 11.9 MB/s eta 0:00:00
  Downloading uvicorn-0.21.0-py3-none-any.whl (57 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.8/57.8 kB 11.6 MB/s eta 0:00:00
  Downloading uvicorn-0.20.0-py3-none-any.whl (56 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 56.9/56.9 kB 12.7 MB/s eta 0:00:00
  Downloading uvicorn-0.19.0-py3-none-any.whl (56 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 56.6/56.6 kB 9.0 MB/s eta 0:00:00
  Downloading uvicorn-0.18.3-py3-none-any.whl (57 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.4/57.4 kB 14.9 MB/s eta 0:00:00
  Downloading uvicorn-0.18.2-py3-none-any.whl (57 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.0/57.0 kB 11.8 MB/s eta 0:00:00
  Downloading uvicorn-0.18.1-py3-none-any.whl (57 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.0/57.0 kB 12.5 MB/s eta 0:00:00
  Downloading uvicorn-0.18.0-py3-none-any.whl (57 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.0/57.0 kB 10.3 MB/s eta 0:00:00
  Downloading uvicorn-0.17.6-py3-none-any.whl (53 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 53.6/53.6 kB 10.1 MB/s eta 0:00:00
  Downloading uvicorn-0.17.5-py3-none-any.whl (53 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 53.6/53.6 kB 7.1 MB/s eta 0:00:00
  Downloading uvicorn-0.17.4-py3-none-any.whl (52 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 52.6/52.6 kB 9.1 MB/s eta 0:00:00
  Downloading uvicorn-0.17.3-py3-none-any.whl (52 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 52.6/52.6 kB 11.1 MB/s eta 0:00:00
  Downloading uvicorn-0.17.2-py3-none-any.whl (52 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 52.7/52.7 kB 11.0 MB/s eta 0:00:00
  Downloading uvicorn-0.17.1-py3-none-any.whl (54 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54.7/54.7 kB 10.7 MB/s eta 0:00:00
  Downloading uvicorn-0.17.0.post1-py3-none-any.whl (54 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54.9/54.9 kB 10.5 MB/s eta 0:00:00
  Downloading uvicorn-0.16.0-py3-none-any.whl (54 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54.9/54.9 kB 14.5 MB/s eta 0:00:00
  Downloading uvicorn-0.15.0-py3-none-any.whl (54 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54.4/54.4 kB 8.7 MB/s eta 0:00:00
  Downloading uvicorn-0.14.0-py3-none-any.whl (50 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 50.5/50.5 kB 8.9 MB/s eta 0:00:00
  Downloading uvicorn-0.13.4-py3-none-any.whl (46 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 46.1/46.1 kB 9.5 MB/s eta 0:00:00
Collecting h11>=0.8
  Downloading h11-0.14.0-py3-none-any.whl (58 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.3/58.3 kB 7.8 MB/s eta 0:00:00
Collecting uvicorn
  Downloading uvicorn-0.13.3-py3-none-any.whl (45 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 45.7/45.7 kB 10.4 MB/s eta 0:00:00
  Downloading uvicorn-0.13.2-py3-none-any.whl (45 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 45.5/45.5 kB 10.9 MB/s eta 0:00:00
  Downloading uvicorn-0.13.1-py3-none-any.whl (45 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 45.5/45.5 kB 11.0 MB/s eta 0:00:00
  Downloading uvicorn-0.13.0-py3-none-any.whl (45 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 45.7/45.7 kB 9.9 MB/s eta 0:00:00
  Downloading uvicorn-0.12.3-py3-none-any.whl (45 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 45.7/45.7 kB 11.6 MB/s eta 0:00:00
  Downloading uvicorn-0.12.2-py3-none-any.whl (45 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 45.1/45.1 kB 11.3 MB/s eta 0:00:00
  Downloading uvicorn-0.12.1-py3-none-any.whl (44 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 44.6/44.6 kB 9.1 MB/s eta 0:00:00
  Downloading uvicorn-0.12.0-py3-none-any.whl (44 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 44.6/44.6 kB 7.2 MB/s eta 0:00:00
  Downloading uvicorn-0.11.8-py3-none-any.whl (43 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 43.3/43.3 kB 7.6 MB/s eta 0:00:00
Collecting h11<0.10,>=0.8
  Downloading h11-0.9.0-py2.py3-none-any.whl (53 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 53.6/53.6 kB 12.1 MB/s eta 0:00:00
Collecting websockets==8.*
  Downloading websockets-8.1-cp38-cp38-manylinux2010_x86_64.whl (78 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.4/78.4 kB 12.4 MB/s eta 0:00:00
Collecting uvicorn
  Downloading uvicorn-0.11.7-py3-none-any.whl (43 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 43.4/43.4 kB 8.1 MB/s eta 0:00:00
  Downloading uvicorn-0.11.6-py3-none-any.whl (43 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 43.2/43.2 kB 7.1 MB/s eta 0:00:00
  Downloading uvicorn-0.11.5-py3-none-any.whl (43 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 43.1/43.1 kB 5.5 MB/s eta 0:00:00
  Downloading uvicorn-0.11.4-py3-none-any.whl (43 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 43.1/43.1 kB 6.4 MB/s eta 0:00:00
  Downloading uvicorn-0.11.3-py3-none-any.whl (42 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.3/42.3 kB 6.5 MB/s eta 0:00:00
  Downloading uvicorn-0.11.2-py3-none-any.whl (42 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.3/42.3 kB 7.9 MB/s eta 0:00:00
Collecting uvloop>=0.14.0
  Downloading uvloop-0.17.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.6/4.6 MB 99.1 MB/s eta 0:00:00
Collecting uvicorn
  Downloading uvicorn-0.11.1-py3-none-any.whl (42 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.2/42.2 kB 9.8 MB/s eta 0:00:00
  Downloading uvicorn-0.11.0-py3-none-any.whl (42 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.2/42.2 kB 7.8 MB/s eta 0:00:00
  Downloading uvicorn-0.10.9-py3-none-any.whl (42 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.5/42.5 kB 8.1 MB/s eta 0:00:00
  Downloading uvicorn-0.10.8.tar.gz (27 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.10.7.tar.gz (27 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.10.6.tar.gz (28 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.10.5.tar.gz (27 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.10.4.tar.gz (27 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.10.3.tar.gz (27 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.10.2.tar.gz (27 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.10.1.tar.gz (27 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.10.0.tar.gz (26 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.9.1.tar.gz (25 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.9.0.tar.gz (24 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.8.6.tar.gz (24 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.8.5.tar.gz (24 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.8.4.tar.gz (24 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.8.3.tar.gz (24 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.8.2.tar.gz (24 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.8.1.tar.gz (24 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.8.0.tar.gz (24 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.7.3.tar.gz (24 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.7.2.tar.gz (24 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.7.1.tar.gz (24 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.7.0.tar.gz (24 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.6.1.tar.gz (24 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.6.0.tar.gz (24 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.5.2.tar.gz (23 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Downloading uvicorn-0.5.1.tar.gz (23 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting websockets>=6.0
  Downloading websockets-11.0.3-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (130 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 130.2/130.2 kB 26.2 MB/s eta 0:00:00
Collecting httptools
  Downloading httptools-0.6.0-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (441 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 441.7/441.7 kB 54.0 MB/s eta 0:00:00
Collecting sniffio>=1.1
  Downloading sniffio-1.3.0-py3-none-any.whl (10 kB)
Collecting pycparser
  Downloading pycparser-2.21-py2.py3-none-any.whl (118 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 118.7/118.7 kB 17.6 MB/s eta 0:00:00
Building wheels for collected packages: genologics, petname, atomicwrites, urlobject, uvicorn
  Building wheel for genologics (setup.py): started
  Building wheel for genologics (setup.py): finished with status 'done'
  Created wheel for genologics: filename=genologics-0.4.6-py3-none-any.whl size=33400 sha256=a155d83a389f2b6988eec319894c03e915916fc4f83df47f2cb6dc376f059d38
  Stored in directory: /home/sebastian.allard/.cache/pip/wheels/17/e1/38/c5b042119360d5e1b8e2f3d16432414d79abb9c382ca3c0a4e
  Building wheel for petname (setup.py): started
  Building wheel for petname (setup.py): finished with status 'done'
  Created wheel for petname: filename=petname-2.6-py3-none-any.whl size=8121 sha256=5369a4db8e6bbbc8505df8973b3ef2d6fdc1084e4af8294b73cca614ae5476e0
  Stored in directory: /home/sebastian.allard/.cache/pip/wheels/df/d0/f7/5bb2e3528016cf736839ec867275e2ecc9c01e0e26c7596fe3
  Building wheel for atomicwrites (setup.py): started
  Building wheel for atomicwrites (setup.py): finished with status 'done'
  Created wheel for atomicwrites: filename=atomicwrites-1.4.1-py2.py3-none-any.whl size=6943 sha256=55e2086981af72a26441c5ab733b0a9868801eabf4e1ce3fd0ae9eae9a14553b
  Stored in directory: /home/sebastian.allard/.cache/pip/wheels/ce/24/c1/56b1ca77dda9577b75eb88d7ec50425f103ec2f6170c6e97df
  Building wheel for urlobject (setup.py): started
  Building wheel for urlobject (setup.py): finished with status 'done'
  Created wheel for urlobject: filename=URLObject-2.4.3-py3-none-any.whl size=14510 sha256=d55f78aa6dfea20fdddba8e043c700f5ba6516bc4d45173f5888a53399b21429
  Stored in directory: /home/sebastian.allard/.cache/pip/wheels/6d/9f/65/f0e414e2ac1be3f726f12d3952e38a624629b1056d5fc76401
  Building wheel for uvicorn (setup.py): started
  Building wheel for uvicorn (setup.py): finished with status 'done'
  Created wheel for uvicorn: filename=uvicorn-0.5.1-py3-none-any.whl size=35998 sha256=a51f888b4ebf11b0d51560ad2c32ce0804ceec36fc892b72d2c0c2aded007cbc
  Stored in directory: /home/sebastian.allard/.cache/pip/wheels/17/3c/02/aa0f1a65bee14cdc952654fdb6fe2bb4f324d80bea1eb89476
Successfully built genologics petname atomicwrites urlobject uvicorn
Installing collected packages: wcwidth, urlobject, pytz, petname, dataclasses, Click, werkzeug, websockets, uvloop, urllib3, tzdata, typing-extensions, tomli, tabulate, SQLAlchemy, sniffio, six, pyyaml, python-multipart, pymysql, pycparser, pyasn1, py, pluggy, oauthlib, numpy, msgpack, more-itertools, markupsafe, lxml, itsdangerous, iniconfig, importlib-resources, idna, httptools, h11, exceptiongroup, et-xmlfile, dnspython, coverage, coloredlogs, colorama, charset-normalizer, certifi, cachetools, blinker, bcrypt, attrs, atomicwrites, wtforms, uvicorn, rsa, requests, python-dateutil, pytest, pydantic-core, pyasn1-modules, openpyxl, Mako, Jinja2, gunicorn, email-validator, cffi, anyio, annotated-types, starlette, requests-oauthlib, pytest-cov, pynacl, pydantic, pandas, housekeeper, google-auth, genologics, Flask, cryptography, cachecontrol, alembic, paramiko, flask-wtf, Flask-Dance, Flask-CORS, Flask-Admin, fastapi, cgmodels, sendmail-container
Successfully installed Click-6.7 Flask-1.1.4 Flask-Admin-1.6.1 Flask-CORS-4.0.0 Flask-Dance-5.1.0 Jinja2-2.11.3 Mako-1.2.4 SQLAlchemy-1.3.24 alembic-1.12.0 annotated-types-0.5.0 anyio-3.7.1 atomicwrites-1.4.1 attrs-23.1.0 bcrypt-4.0.1 blinker-1.6.2 cachecontrol-0.13.1 cachetools-5.3.1 certifi-2023.7.22 cffi-1.16.0 cgmodels-0.11.0 charset-normalizer-3.3.0 colorama-0.4.6 coloredlogs-15.0.1 coverage-7.3.2 cryptography-41.0.4 dataclasses-0.6 dnspython-2.4.2 email-validator-2.0.0.post2 et-xmlfile-1.1.0 exceptiongroup-1.1.3 fastapi-0.103.2 flask-wtf-1.2.1 genologics-0.4.6 google-auth-2.23.2 gunicorn-21.2.0 h11-0.14.0 housekeeper-4.8.5 httptools-0.6.0 idna-3.4 importlib-resources-6.1.0 iniconfig-2.0.0 itsdangerous-1.1.0 lxml-4.9.3 markupsafe-2.0.1 more-itertools-10.1.0 msgpack-1.0.7 numpy-1.24.4 oauthlib-3.2.2 openpyxl-3.0.10 pandas-2.0.3 paramiko-3.3.1 petname-2.6 pluggy-1.3.0 py-1.11.0 pyasn1-0.5.0 pyasn1-modules-0.3.0 pycparser-2.21 pydantic-2.4.2 pydantic-core-2.10.1 pymysql-1.1.0 pynacl-1.5.0 pytest-7.4.2 pytest-cov-4.1.0 python-dateutil-2.8.2 python-multipart-0.0.6 pytz-2023.3.post1 pyyaml-6.0.1 requests-2.31.0 requests-oauthlib-1.3.1 rsa-4.9 sendmail-container-0.2.0 six-1.16.0 sniffio-1.3.0 starlette-0.27.0 tabulate-0.9.0 tomli-2.0.1 typing-extensions-4.8.0 tzdata-2023.3 urllib3-2.0.6 urlobject-2.4.3 uvicorn-0.5.1 uvloop-0.17.0 wcwidth-0.2.8 websockets-11.0.3 werkzeug-1.0.1 wtforms-3.0.1
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/sebastian.allard
done.
Log deploy... done.
cg, version 51.2.2
```

seallard commented:
Tested with
```
θ67° [sebastian.allard@hasta:/home/proj/stage/demultiplexed-runs] [S_base] $ cg demultiplex finish flow-cell 230824_A00187_0994_BHGW2VDRX3
Running cg demultiplex.
Finish flow cell 230824_A00187_0994_BHGW2VDRX3
Flow cell 230824_A00187_0994_BHGW2VDRX3 will be skipped: Sample sheet /home/proj/stage/housekeeper-bundles/HGW2VDRX3/2023-08-24/SampleSheet.csv does not exist in housekeeper.
```

seallard commented:
Deployed cg and housekeeper to Hasta and tested them.
Deployed to cg-vm1 and tested to place an order, the status db view still works.

henrikstranneheim reviewed (approved):
💯

seallard commented:
Upgrade Python in stage. Deployed this branch. Deployed corresponding branch in housekeeper. Tested basic commands.

seallard commented:
Upgrade Python version in `P_main` and deployed master to prod, as well as the housekeeper master branch.
